ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -118,6 +118,15 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 			clusterCRs = append(clusterCRs, crValue)
 		} else if mapSourceCR["kind"] == "BareMetalHost" || mapSourceCR["kind"] == "NMStateConfig" {
 			for ndId := range siteConfigTemp.Spec.Clusters[clusterId].Nodes {
+				if mapSourceCR["kind"] == "BareMetalHost" &&
```

Comment:
the loop will re-create the configMap for every node even if all of them use the same configMap defined in the cluster level. I guess better to check only for the node here and then level up the cluster check out of the loop.